runelite-client/src/main/java/net/runelite/client/game/AgilityShortcut.java

```diff
@@ -243,22 +251,42 @@
 	@Getter
 	private final int[] obstacleIds;
 
-	AgilityShortcut(int level, String description, WorldPoint mapLocation, WorldPoint worldLocation, int... obstacleIds)
+	AgilityShortcut(List<Requirement> requirements, String description, WorldPoint mapLocation, WorldPoint worldLocation, int... obstacleIds)
 	{
-		this.level = level;
+		this.requirements = requirements;
 		this.description = description;
 		this.worldMapLocation = mapLocation;
 		this.worldLocation = worldLocation;
 		this.obstacleIds = obstacleIds;
 	}
 
+	AgilityShortcut(int level, String description, WorldPoint mapLocation, WorldPoint worldLocation, int... obstacleIds)
+	{
+		this(ImmutableList.of(new SkillRequirement(Skill.AGILITY, level)), description, mapLocation, worldLocation, obstacleIds);
+	}
+
 	AgilityShortcut(int level, String description, WorldPoint location, int... obstacleIds)
 	{
-		this(level, description, location, location, obstacleIds);
+		this(ImmutableList.of(new SkillRequirement(Skill.AGILITY, level)), description, location, location, obstacleIds);
+	}
+
+	AgilityShortcut(int level, Diary diaryRequirement, String description, WorldPoint location, int... obstacleIds)
+	{
+		this(ImmutableList.of(new SkillRequirement(Skill.AGILITY, level), new AchievementDiaryRequirement(diaryRequirement)), description, location, location, obstacleIds);
+	}
+
+	public boolean satisfiesAll(Client client)
+	{
+		boolean satisfies = true;
+		for (Requirement requirement : requirements)
+		{
+			satisfies &= requirement.isSatisfied(client);
+		}
+		return satisfies;
```

Comment:
Don't loop through every requirement if an unsatisfied one is encountered early.
```suggestion
		for (Requirement requirement : requirements)
		{
			if (!requirement.isSatisfied(client))
			{
				return false;
			}
		}
		return true;
```